interface/src/app/node/node.js

```diff
@@ -92,5 +92,223 @@ angular.module('adage.node', [
   }]
 )
 
+.directive('highRangeExp', ['$http', '$log', function($http, $log) {
+  return {
+    templateUrl: 'node/high_range_exp.tpl.html',
+    restrict: 'E',
+    scope: {
+      nodeId: '@',
+      inputTopNum: '@topExp'
+    },
+    link: function($scope) {
+      $scope.queryStatus = 'Connecting to the server ...';
```

Comment:
When defining a directive, we have two choices: a link function or a controller. I personally prefer using link function when possible. Since the directive here has its own template html file and isolated scope is used, any expression in the template html will always refer to the directive's own scope, **controllerAs**'s advantage of explicit scope is not so important (IMHO).